### ISSUE TITLE
Error while downloading the updated schema.graphqls using the command given in the documentation

### ISSUE THREAD

**Issue by sam43** [apollographql/apollo-kotlin#4539]
### Summary
I have been facing some issue while downloading the `schema.graphqls` file using the process mentioned in [this doc](https://www.apollographql.com/docs/kotlin/tutorial/02-add-the-graphql-schema) and Looks like the process doesn't work with gradle version 7.4 or so.

### Version
I am currently using the latest version for the process which is `const val apolloVersion = "3.7.1"`. But I downloaded the schema beforehand.

### Logs

> sayem@As-MacBook-Pro Mobile_Android_MVA % `./gradlew downloadApolloSchema --endpoint='https://xxx-api.xxx.com/graphql' --schema=connections/src/main/graphql/schema.graphqls`
> 
> Welcome to Gradle 7.4!
> 
> Here are the highlights of this release:
>  - Aggregated test and JaCoCo reports
>  - Marking additional test source directories as tests in IntelliJ
>  - Support for Adoptium JDKs in Java toolchains
> 
> For more details see https://docs.gradle.org/7.4/release-notes.html
> 
> Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Unable to start the daemon process.
> This problem might be caused by incorrect configuration of the daemon.
> For example, an unrecognized jvm option is used.
> Please refer to the User Manual chapter on the daemon at https://docs.gradle.org/7.4/userguide/gradle_daemon.html
> Process command line: /opt/homebrew/Cellar/openjdk/19.0.1/libexec/openjdk.jdk/Contents/Home/bin/java -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.base/java.nio.charset=ALL-UNNAMED --add-opens java.base/java.net=ALL-UNNAMED --add-opens java.base/java.util.concurrent.atomic=ALL-UNNAMED -Xmx5120m -Dfile.encoding=UTF-8 -Duser.country=BD -Duser.language=en -Duser.variant -cp /Users/sayem/.gradle/wrapper/dists/gradle-7.4-bin/c0gwcg53nkjbqw7r0h0umtfvt/gradle-7.4/lib/gradle-launcher-7.4.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 7.4
> Please read the following process output to find out more:
> -----------------------
> Unrecognized VM option 'MaxPermSize=512m'
> Error: Could not create the Java Virtual Machine.
> Error: A fatal exception has occurred. Program will exit.
> 
> 
> * Try:
> > Run with --stacktrace option to get the stack trace.
> > Run with --info or --debug option to get more log output.
> > Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
<img width="1265" alt="Screenshot 2022-11-30 at 1 42 00 PM" src="https://user-images.githubusercontent.com/4275505/204737627-c9c9cb18-6017-48da-9e99-446e6a8690fe.png">

### Steps to reproduce the behaviour
Add context about the problem here. How it happened and how to reproduce it:  
* [Download the project](https://github.com/sam43/apollo-kotlin-tutorial/tree/apollo-v3-http-cache-test) and run the command to download the `schema.graphqls`
* This command:  `./gradlew :app:downloadApolloSchema --endpoint='https://apollo-fullstack-tutorial.herokuapp.com/graphql' --schema=app/src/main/graphql/com/example/rocketreserver/schema.graphqls`
* I have already tried and got [this](https://imgur.com/a/T4w4bp0).

**Comment by martinbonnin** [comment]
Hi 👋 Thanks for sending this!

It looks like it is related to your Java install. 

> Unrecognized VM option 'MaxPermSize=512m'

`MaxPermSize` is an older CLI option that is not supported in newer Java versions

> JAVA_HOME is set to an invalid directory

You'll need to set JAVA_HOME to the directory where your JDK is installed, not to the `java` binary itself.

I recommend installing Java from [Homebrew](https://brew.sh/) if you are on a mac: https://twitter.com/JakeWharton/status/1463524757765251082

Alternatively, if you're on Windows or Linux, [SdkMan](https://sdkman.io/) has also some JDK distributions available.

Let us know how that works!

**Comment by sam43** [comment]
Hi @martinbonnin , Thank you for your swift response to the ticket. I installed the `openjdk` via homebrew after creating the ticket. the path that the command `whereis java` is showing is wrong, I had to manually override the path of **JAVA_HOME** and it's working now. 
For those who will be facing the same in the future I'm sharing the default openjdk dir/location on mac. 
Add this following line to your `~/.zshrc` or `~/.bash_profile` (system env) file to avoid this issue.

@martinbonnin thanks again fir your support. 👍🏻
